packages/yew-router/src/utils.rs

```diff
@@ -1,10 +1,25 @@
+use std::cell::RefCell;
 use wasm_bindgen::JsCast;
 
 pub(crate) fn strip_slash_suffix(path: &str) -> &str {
     path.strip_suffix("/").unwrap_or(&path)
 }
 
+static BASE_URL_LOADED: std::sync::Once = std::sync::Once::new();
+thread_local!(static BASE_URL: RefCell<Option<String>> = RefCell::new(None));
+
+// This exists so we can cache the base url. It costs us a `to_string` call instead of a DOM API call.
+// Considering base urls are generally short, it *should* be less expensive.
 pub fn base_url() -> Option<String> {
+    BASE_URL_LOADED.call_once(|| {
+        BASE_URL.with(|val| {
+            *val.borrow_mut() = fetch_base_url();
+        })
+    });
+    BASE_URL.with(|it| it.borrow().as_ref().map(|it| it.to_string()))
```

Comment:
Why not use an option instead of using both an additional once?

Reply:
`fetch_base_url` needs to called only once from this function at the very start to populate state. `Option` can't tell the difference between uninitialized and no route found. In fact, `Option` is used currently for the latter case.